### PR TITLE
Travis use conda itk on OSX

### PR DESCRIPTION
**Pull request by KrisThielemans**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -130,6 +129,18 @@ before_install:
...
+      if [[ $EXTRA_BUILD_FLAGS == *"-DUSE_ITK=ON"* ]] && [[ $EXTRA_BUILD_FLAGS == *"-DUSE_SYSTEM_ITK=ON"* ]]; then
+        echo "Installing ITK via miniconda"
+        wget https://repo.continuum.io/miniconda/Miniconda$PYMVER-latest-MacOSX-x86_64.sh -O miniconda.sh;
+        bash miniconda.sh -b -p $HOME/miniconda
+        export PATH="$HOME/miniconda/bin:$PATH"
+        hash -r
+        conda config --set always_yes yes --set changeps1 no
+        conda update -q conda
+        conda install -c conda-forge libitk
+        EXTRA_BUILD_FLAGS="$EXTRA_BUILD_FLAGS -DITK_DIR=$HOME/miniconda/pkgs/itk-4.12.0-0/lib/cmake/ITK-4.12"
```
Comment (KrisThielemans) [inline .travis.yml]: this is incorrect as conda is now at libitk-4.13.0. It seems in any case that the line is not needed as CMake is picking it up, as I see in the Travis log file a mention of `/Users/travis/miniconda/include/ITK-4.13/itkImage.h`

**Comment by KrisThielemans** [comment]
Travis failure ([log](https://api.travis-ci.org/v3/job/480922636/log.txt))
```
make[5]: *** No rule to make target `/Users/travis/miniconda3/conda-bld/libitk_1546796752855/_build_env/lib/libz.dylib', needed by `src/xSTIR/pSTIR/_pystir.so'.  Stop.
```
This looks to me like a problem in the conda build. Easiest to check in a local build of course. I think that the `itk`  target depends on this file (as opposed to an installed `zlib`). It could be via its hdf5 dependency. To verify, @rijobro could you check on your machine with something similar to
```
grep libz /wherever/cmakeITK-4.13/*.*
grep libz /wherever/cmakeITK-4.13/Modules/*
```

**Comment by KrisThielemans** [comment]
I notice that `conda libitk` also installs `hdf5-1.10.4` and `python-2.7.15`. that'll be interesting for CMake... (it seems it's still preferring the system python).
`

**Comment by rijobro** [comment]
doing `grep libz ~/miniconda3/lib/cmake/ITK-4.13/*.*` gives:
```
/Users/rich/miniconda3/lib/cmake/ITK-4.13/ITKTargets-release.cmake:  IMPORTED_LINK_INTERFACE_LIBRARIES_RELEASE "/Users/travis/miniconda3/conda-bld/libitk_1546796752855/_build_env/lib/libz.dylib"
/Users/rich/miniconda3/lib/cmake/ITK-4.13/ITKTargets-release.cmake:  IMPORTED_LINK_INTERFACE_LIBRARIES_RELEASE "/Users/rich/miniconda3/lib/libhdf5_cpp.dylib;/Users/rich/miniconda3/lib/libhdf5.dylib;/usr/lib/libpthread.dylib;/Users/rich/miniconda3/lib/libz.dylib;/usr/lib/libdl.dylib;/usr/lib/libm.dylib;/Users/travis/miniconda3/conda-bld/libitk_1546796752855/_build_env/lib/libz.dylib;m;dl"
/Users/rich/miniconda3/lib/cmake/ITK-4.13/ITKTargets.cmake:  INTERFACE_LINK_LIBRARIES "/Users/travis/miniconda3/conda-bld/libitk_1546796752855/_build_env/lib/libz.dylib"
/Users/rich/miniconda3/lib/cmake/ITK-4.13/ITKTargets.cmake:  INTERFACE_LINK_LIBRARIES "/Users/travis/miniconda3/conda-bld/libitk_1546796752855/_build_env/lib/libz.dylib"
```

and doing `grep libz ~/miniconda3/lib/cmake/ITK-4.13/Modules/*.*` gives:
```
/Users/rich/miniconda3/lib/cmake/ITK-4.13/Modules/ITKHDF5.cmake:set(ITKHDF5_LIBRARIES "/Users/rich/miniconda3/lib/libhdf5_cpp.dylib;/Users/rich/miniconda3/lib/libhdf5.dylib;/usr/lib/libpthread.dylib;/Users/rich/miniconda3/lib/libz.dylib;/usr/lib/libdl.dylib;/usr/lib/libm.dylib")
/Users/rich/miniconda3/lib/cmake/ITK-4.13/Modules/ITKPNG.cmake:set(ITKPNG_LIBRARIES "/Users/rich/miniconda3/lib/libpng.dylib;/Users/travis/miniconda3/conda-bld/libitk_1546796752855/_build_env/lib/libz.dylib")
/Users/rich/miniconda3/lib/cmake/ITK-4.13/Modules/ITKZLIB.cmake:set(ITKZLIB_LIBRARIES "/Users/travis/miniconda3/conda-bld/libitk_1546796752855/_build_env/lib/libz.dylib")
```

I'm not quite sure what to make of this. Especially `/Users/travis/...` because this is on my machine and none of the commands have used the word `travis`...

I also suspect that `conda-bld/` is a temporary folder because it no longer exists on my machine.

However, `/Users/rich/miniconda3/lib/libz.dylib` and `/Users/rich/miniconda3/lib/libhdf5_cpp.dylib` both appear to be symlinks in the correct location.

**Comment by paskino** [comment]
You are right `conda-bld` is a temporary directory where conda creates a new environment for the build and test

**Comment by rijobro** [comment]
Waiting on Travis, but locally at least it looks like downloading an offline version (discussed here: conda-forge/libitk-feedstock#24) solves the problem.

**Comment by KrisThielemans** [comment]
@rijobro, try doing
```sh
DYLD_PRINT_LIBRARIES=YES ctest -VV
```
[see here](https://apple.stackexchange.com/questions/247227/how-to-find-which-shared-library-is-loaded-by-a-process-on-osx)

**Comment by rijobro** [comment]
Gave this a go, but currently blocked by #300.

**Comment by rijobro** [comment]
The verbose Travis log is here: https://api.travis-ci.org/v3/job/488963752/log.txt

I can't make much sense of it. As far as I can see, none of the libraries are printed prior to `PET_TESTS_PYTHON`.

**Comment by rijobro** [comment]
There is a problem with @rpaths (described here: https://github.com/conda-forge/libitk-feedstock/issues/24). 

Locally, I managed to solve this by linking all the necessary libraries (via trial and error) to another location, and adding that location do the `DYLD_LIBRARY_PATH`. For some reason, this isn't working on Travis, and the error message isn't very helpful:

```
2: ======================================================================

2: ERROR: Failure: ImportError (No module named _pystir)

2: ----------------------------------------------------------------------

2: Traceback (most recent call last):

2:   File "/Users/travis/Library/Python/2.7/lib/python/site-packages/nose/loader.py", line 418, in loadTestsFromName

2:     addr.filename, addr.module)

2:   File "/Users/travis/Library/Python/2.7/lib/python/site-packages/nose/importer.py", line 47, in importFromPath

2:     return self.importFromDir(dir_path, fqname)

2:   File "/Users/travis/Library/Python/2.7/lib/python/site-packages/nose/importer.py", line 94, in importFromDir

2:     mod = load_module(part_fqname, fh, filename, desc)

2:   File "/Users/travis/build/CCPPETMR/SIRF-SuperBuild/sources/SIRF/src/xSTIR/pSTIR/tests/tests_three.py", line 16, in <module>

2:     from pSTIR import *

2:   File "/Users/travis/build/CCPPETMR/SIRF-SuperBuild/INSTALL/python/pSTIR/__init__.py", line 1, in <module>

2:     from sirf.STIR import *

2:   File "/Users/travis/build/CCPPETMR/SIRF-SuperBuild/INSTALL/python/sirf/STIR.py", line 41, in <module>

2:     import sirf.pystir as pystir

2:   File "/Users/travis/build/CCPPETMR/SIRF-SuperBuild/INSTALL/python/sirf/pystir.py", line 17, in <module>

2:     _pystir = swig_import_helper()

2:   File "/Users/travis/build/CCPPETMR/SIRF-SuperBuild/INSTALL/python/sirf/pystir.py", line 16, in swig_import_helper

2:     return importlib.import_module('_pystir')

2:   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module

2:     __import__(name)

2: ImportError: No module named _pystir
```

**Comment by KrisThielemans** [comment]
this might be resolved now with updates in `libitk-dev`. Can give this a go again.

**Comment by rijobro** [comment]
I don't know how to fix this one. It works locally on my OSX, `ls $HOME/miniconda/lib/cmake/ITK-4.13` contains the expected files such as `ITKConfig.cmake` and my SuperBuild is happy to use it.

**Comment by KrisThielemans** [comment]
seems conda failed to update
https://travis-ci.org/github/SyneRBI/SIRF/jobs/683301530#L4210-L4212

by the way, it seems to go to python 2.7.

Anyway, I feel that mixing conda and brew is probably a bad idea.

**Comment by KrisThielemans** [comment]
obsolete

**Pull request closed without merging**